.github/workflows/update_libs.yaml

```
@@ -0,0 +1,54 @@
+# reusable workflow for checking library versions and opening PRs to bump
+name: Update Charm Libraries
+
+on:
+  workflow_call:
+    inputs:
+      charmcraft_channel:
+        description: Channel from which to install charmcraft
+        default: 'latest/candidate'
+        required: false
+        type: string
+    secrets:
+      CHARMCRAFT_CREDENTIALS:
+        required: true
+  workflow_dispatch:
+    inputs:
+      charmcraft_channel:
+        description: Channel from which to install charmcraft
+        default: 'latest/candidate'
+        required: false
+        type: string
+
+jobs:
+  update-libs:
+    name: Update charm libraries
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - run: sudo snap install charmcraft --channel ${{ inputs.charmcraft_channel }} --classic
+
+      - name: update charm libs
+        run: charmcraft fetch-lib
+        env:
+          CHARMCRAFT_AUTH: ${{ secrets.CHARMCRAFT_CREDENTIALS }}
+
+      - name: Create Pull Request
+        id: create-pull-request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          title: '(Automated) Update Charm Libs'
+          body: 'Update charm libs'
+          commit-message: 'Update charm libs'
+          signoff: false
```

Adding this should solve the issue of github-actions pushing unsigned commits on our behalf:
```suggestion
          token: ${{ secrets.GITHUB_TOKEN }}
          commit-message: "chore: update charm libraries"
          committer: "Github Actions <github-actions@github.com>"
          author: "Github Actions <github-actions@github.com>"
```